### PR TITLE
release-notes.txt: add 2024.01 release notes

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

It is that time again!  Please add changes and comments as needed!

### Testing procedure

Read it!

### CODE REVIEW

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,528 @@
...
+
+* dist/tools/uf2: add target to also copy families.json file (#20042)
+* backport_pr.py: increase entries per page for branches GET (#19985)
+* boards/esp32s3-usb-otg: fix the table of contents in documentation (#20156)
+* boards/native: only use pyterm wrapper with term target (#20264)
+* boards/nucleo-f413zh: Fix DMA mapping (#20225)
+* ci: fix murdock tests after dwq change (#20268)
+* cpu/cc2538: fix EUI provider (#20228)
+* cpu/cortexm_common: work around bug on WFI for STM32 (#20149)
+* cpu/nrf5x i2c: Set up correct interrupts after NOSTOP transmissions (#20282)
```
Comment (chrysn) [inline release-notes.txt:241]:
```suggestion
```

All that this fix did was replace a lockup situation with a race condition. The next release will contain #20299 and #20298 which fix things properly, let's not make people retry anything that didn't work before on nRF52 I2C yet.

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,528 @@
...
++ boards/generic-cc2538-cc2592-dk: add board support (#20226)
++ boards/nucleo-l4r5zi : add PWM configuration (#20036)
++ boards/nucleo-l552ze: add PWM configuration (#20032)
++ boards/p-nucleo-wb55: add PWM configuration (#20120)
++ boards: add support for Adafruit Feather nRF52840 Sense (#20027)
++ drivers/include/periph: add FREQM peripheral driver (#20020)
++ drivers/ws281x: Add gpio_ll and timer based driver (#19891)
++ pyterm: add native support (#20172)
++ sys/shell: add firmware version to version cmd (#20244)
+* AVR-8: Implement Power Management (#19784)
```
Comment (maribu) [inline release-notes.txt:103]: This might be one of the bigger changes to highlight.

**Comment by MrKevinWeiss** [comment]
ping @RIOT-OS/maintainers

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,528 @@
...
+
+Documentation
+-------------
+
+* boards/nucleo-l433rc: doc update - image and MCU table (#20098)
+* boards/nucleo-l452re: doc update (#20207)
+* boards/p-nucleo-wb55: documentation update (#20115)
+* doc/mainpage: don't claim to be a microkernel (#20246)
+* doc: Extend and take responsibility for 802.15.4 security roadmap (#20123)
+* doc: Move roadmap in from the wiki (#20122)
```
Comment (chrysn) [inline release-notes.txt:156]:
```suggestion
* doc: Move and update roadmap in from the wiki (#20122, #20123)
```

The latter justifies why it says "update" there, but standalone the title only makes sense in a PR context where the author is visible, but has no business as a standalone entry in change logs.

File: release-notes.txt [hunk 4]
```diff
@@ -1,3 +1,528 @@
...
++ cpu/msp430: improve periph_timer (#20160)
++ cpu/stm32/periph: add low-level SDMMC peripheral driver (#19540)
++ cpu/atmega_common: implement periph_timer_query_freqs (#20142)
++ cpu/cc26xx_cc13xx: implement periph_timer_query_freqs (#20143)
++ cpu/kinetis: implement periph_timer_query_freqs (#20144)
++ cpu/nrf5x_common: implement periph_timer_query_freqs (#20145)
++ cpu/qn908x: implement periph_timer_query_freqs (#20146)
++ cpu/sam0_common: implement periph_timer_query_freqs (#20147)
++ cpu/stm32: add ADCAL operation (#20044)
++ cpu/stm32: implement periph_timer_query_freqs (#20148)
```
Comment (maribu) [inline release-notes.txt:129]: Theses `periph_timer_query_freqs` changes may also be something to highlight.

Comment (chrysn) [reply release-notes.txt:129]: They're also something that could be grouped in the change log.

**Review by mguetschow — DISMISSED** [review]
LGTM otherwise.

**Comment by chrysn** [comment]
>     * groundwork for 64 bit support --> expect support for `native64` and maybe first RISC-V 64 bit boards in next release

That's noteworthy from an inside PoV, but for the audience of release notes (whose priorities are more "what can it do now that it couldn't before"), I think it is not relevant for a spotlight item in this release.

**Comment by MrKevinWeiss** [comment]
OK. added the changes!

File: release-notes.txt [hunk 5]
```diff
@@ -1,3 +1,527 @@
...
++ boards/feather-nrf52840-sense: Add bootloader info (#20034)
++ boards/gd32vf103c-start: new board (#20270)
++ boards/generic-cc2538-cc2592-dk: add board support (#20226)
++ boards/nucleo-l4r5zi : add PWM configuration (#20036)
++ boards/nucleo-l552ze: add PWM configuration (#20032)
++ boards/p-nucleo-wb55: add PWM configuration (#20120)
++ boards: add support for Adafruit Feather nRF52840 Sense (#20027)
++ drivers/include/periph: add FREQM peripheral driver (#20020)
++ drivers/ws281x: Add gpio_ll and timer based driver (#19891)
++ pyterm: add native support (#20172)
```
Comment (benpicco) [inline release-notes.txt:101]: This is also a user-visible change

File: release-notes.txt [hunk 6]
```diff
@@ -1,3 +1,527 @@
...
+* tests/sys/psa_crypto_se_cipher: disable test on esp32-wroom-32 (#20152)
+* tests/sys/psa_crytpo_se: disable test on esp32-wroom-32 (#20150)
+* tests/unittests: blacklist LLVM on native (#19958)
+* tests: Slightly increase coverage of fmt unittests (#20272)
+
+API Changes
+-----------
+
++ print: Added size_t print format specifier (#20194)
+* drivers/mtd: use XFA for pointers to defined MTDs (#19465)
```
Comment (benpicco) [inline release-notes.txt:205]: That's a breaking change that requires the user to adapt their code - the PR contains a short description of what migration steps need to be performed.

File: release-notes.txt [hunk 7]
```diff
@@ -1,3 +1,528 @@
...
+* doc: Extend and take responsibility for 802.15.4 security roadmap (#20123)
+* doc: Move roadmap in from the wiki (#20122)
+
+Build System / Tooling
+----------------------
+
++ dist/testbed-support: Add openmote board (#19979)
++ dist/tools/riotboot_gen_hdr/genhdr: add update command (#20239)
+* Bump version of UF2 package (#20035)
+* examples/gnrc_border_router: enable setting ZEP topology (#20006)
```
[the remark this answers is not in the document]
Comment (benpicco) [reply release-notes.txt:164]: It's just a small patch that adds the convenience of selecting the topology file from the border router Makefile

File: release-notes.txt [hunk 8]
```diff
@@ -1,3 +1,527 @@
...
+Examples
+--------
+
+* examples/dtls-sock: Enable non 32-bit architectures (#20196)
+* examples/dtls-sock: make use of helper functions (#20125)
+
+Testing
+-------
+
++ tests/periph/uart_locate_pins: new test/utility app (#20253)
```
Comment (benpicco) [inline release-notes.txt:183]: This is a neat new tool

File: release-notes.txt [hunk 9]
```diff
@@ -1,3 +1,528 @@
...
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release
+==================
+
+The 2024.01 release includes:
+
+- Breaking change for board definition due to XFA for pointers to defined MTD
```
Comment (benpicco) [inline release-notes.txt:23]: Let's be nice to our users and tell them what thy need to do to fix their boards:

To adapt boards definitions to the new API:
 - remove the `MTD_NUMOF` definition from `board.h`
 - add `MTD_XFA_ADD(<mtd_dev>, <idx>);` to the definition of `<mtd_dev>`.
 - `MTD_0`, `MTD_1`, … defines are no longer needed. Use `mtd_dev_get(0)`, `mtd_dev_get(1)`, … instead.

File: test_configs.py
Comment (maribu) [inline test_configs.py:1]: Whoopsie?

Comment (MrKevinWeiss) [reply test_configs.py:1]: yup, just noticed that.

File: release-notes.txt [hunk 10]
```diff
@@ -1,3 +1,533 @@
...
+
+- Breaking change for board definition due to XFA for pointers to defined MTD.
+  This means users will need to:
+    - remove the `MTD_NUMOF` definition from `board.h`
+    - add `MTD_XFA_ADD(<mtd_dev>, <idx>);` to the definition of `<mtd_dev>`.
+    - `MTD_0`, `MTD_1`, … defines are no longer needed. Use `mtd_dev_get(0)`,
+      `mtd_dev_get(1)`, … instead.
+- Many fixes to peripherals due to the periph testing shield
+- The AVR-8 now has power management
+- The ESP32s now support the SDMMC peripheral driver
```
[the remark this answers is not in the document]
Comment (benpicco) [reply release-notes.txt]: I know, but since there actually *is* a ESP32S that could be misunderstood.

**Review by benpicco — APPROVED** [review]

**Pull request merged**